### DISCUSSION TITLE
How much bandwidth?

### DISCUSSION THREAD

**Discussion by nathanvercaemert** [coder/code-server#3631] [Q&A] [answered]
How much data does this application use? I understand that the computations aren't trivial on the host computer, but I'm more concerned with the connection between the host and client. Could it be managed on a very low speed internet connection? I've searched a little bit without finding any answers. Sorry if I missed something very obvious.

**Comment by jsjoeio** [comment]
This is a fantastic question and to be honest, I'm not sure how to answer it 🤔 

> Could it be managed on a very low speed internet connection?

Assuming you're talking about deploying code-server to a remote VM versus running locally, I think it depends.

@oxy or @code-asher would know how to answer this better.

**Reply by nathanvercaemert** [reply]
Thanks for responding! I'm talking about deploying code-server on one machine (could be a droplet/VM or just a Linux box at home) and then accessing it remotely. For my use case, I would be accessing it with a mobile hotspot, and I wouldn't want to burn through all of my high speed data.

I apologize if my question doesn't make sense because of my lack of understanding of the whole application. Maybe I'm approaching this the wrong way, and I'm mixed up about how computationally intensive code-server is in the first place, and I should just run everything locally instead of having to use a more powerful remote machine.

**Reply by jsjoeio** [reply]
No apology needed! Definitely a valid question. I wish there an easy way to answer this but it's a bit out of my wheelhouse. Let's see what the others say. They should know.

**Comment by code-asher** [comment, marked as answer]
This isn't something we've ever measured so unfortunately I can't give
any hard numbers.

It will depend on the size of the files you are editing, how often you
save, the amount of traffic going to and from the language servers, etc.

It could be interesting to measure this and see what an average use case
looks like.

**Comment by nathanvercaemert** [comment]
Thanks for all of the help here, y'all. Such an active community is definitely encouraging. It sounds like the best approach for me for now might be to make sure I'm comfortable running it locally. I hear about people running code-server on Raspberry Pis, so it can't be too terribly computationally intensive. I was hoping you could adamantly say that it was comparable to the bandwidth required for an SSH connection, but it sounds like it could be considerably more. Thanks again.
